### ISSUE TITLE
Add integration with radarr

### ISSUE THREAD

**Issue by chippoman** [JasonHHouse/gaps#154]
Gaps currently only shows you the missing movie. If you want to add it to another application such as radarr, you need to manually add it.

There would be a configuration screen where you specify your radarr instance ip address, port, and api key. On Gaps' missing page, there would be a link for each missing movie to add it to radarr as wanted movie.

There could also be an "Add all" that would send everything to radarr and also an option where you could go into an "edit" mode, put checkboxes next to the movies you want to add to radarr and a button "Add selected"

**Comment by JasonHHouse** [comment]
That's for pointing this out. I didn't know Radarr had an API like that. I'll check it out.

**Comment by calebpaine** [comment]
Another idea might also be to use Radarr for the DB/Search, since its entirely possible to have a movie missing from Plex, but is already added to Radarr and there currently no available sources.

Using Radarr as a DB might also alleviate those wanting integration with Emby/Jellyfin as well.

**Comment by JasonHHouse** [comment]
Yeah, I hadn't seen the API before. I checked it out and this is a good option, but it will take some time to get everything integrated.

**Comment by FunnyPocketBook** [comment]
Not sure if you are already aware of it since you mentioned "manually add", but you can use gaps' RSS feed to connect it to Radarr and it then automatically pulls the movies from the RSS feed.

https://github.com/JasonHHouse/Gaps/wiki/Radarr-List

**Comment by JasonHHouse** [comment]
@FunnyPocketBook Thanks for pointing that out. Yes, Radarr support currently exists but it is manual at the moment.

**Comment by FunnyPocketBook** [comment]
Wouldn't it be more or less the same "manual/effort" if you were to integrate Radarr's API? With both methods you'd have to put in some information manually.

With the Radarr API you have to get Radarr's API and put it into a field in Gaps, with the RSS feed you need the feed URL and put it into a field in Radarr. Both methods require some tweaking of the options (like which library you want to include, set a limit on how many to add etc.), but at the end of the day you would have done the exact same amount of copy pasting, no?

Of course, if you would add it it would be great (because more features is nice) but right now I just don't see the advantage of implementing that if there is already another method of doing exactly that what OP requested.

**Comment by calebpaine** [comment]
In my mind there's two different uses cases for having UI integration with Radarr vs using the RSS feeds, even though they both accomplish the same end goal.

For me, after running gaps there were over 250 missing movies. If I plugged that into Radarr and let it rip in auto mode, my ISP might send a hit squad after me.

Therefore a UI integrated option where I can pick and choose which missing movies I'd like sent to Radarr would be nice. Further the integration might be able to check/indicate which of my missing movies are already in Radarr and I can then go there and investigate why its missing.

For those with smaller libraries or who just don't care and want to "let 'er rip", yes the RSS connection will (probably) do that just fine.

**Comment by FunnyPocketBook** [comment]
Oh yeah, you are totally right. Apparently Radarr only allows to set a limit on Trakt lists... Until now I have only used Trakt lists so I thought the limit could be set on any kind of list and I am able to "let 'er rip" with a 500 movie Gaps feed. In that case a UI intgrated option makes total sense.

**Comment by bakerboy448** [comment]
@JasonHHouse  I'd suggest keeping in mind the V3 API since V2 is only getting critical bug fixes at this point

**Comment by JasonHHouse** [comment]
Yes, I need to update my local to V3. I'll focus on that first and can add V2 if I have time. Thanks!

**Comment by g4m3r7ag** [comment]
V3 Was just updated so you can manually choose items from the list. When you create it, don't set to auto add. Then under Discover you can create a custom filter for Lists contains and the name of your list, then select which movies to add and choose add. This is how I'm doing it to keep from adding 300+ movies all at once.

**Comment by tuxpeople** [comment]
I did the "Radarr takes RSS from Gaps"-thing in Radarr v3 successfully. It's a little bit different from what is documented for Radarrv2, I'll do a PR asap (can I do PRs for wiki in GitHub? Never done this before. Will find out)

**Comment by tuxpeople** [comment]
Looks like I can't, so @JasonHHouse feel free to add this to the wiki if you think it's helpful:

Go to `Settings` -> `Lists` and add a new list. Instead of choosing `StevenLu` like in Radarr v2, choose `StevenLu Custom` under `Advanced List`. Under `URL`, dass the URL to the Gaps Feed of missing movies. Adjust the different list settings as you like (like `Enable Automatic Add` yes/no, `Search on Add` yes/no etc.), then `Test` and `Save`.

![Bildschirmfoto 2022-01-29 um 17 23 56](https://user-images.githubusercontent.com/1506076/151668803-5638b442-76c0-4f24-b593-818aa40b5048.png)

**Comment by JasonHHouse** [comment]
@tuxpeople Added, thanks for writing that up.

**Comment by pk-1966** [comment]
What about an alternative enhancement in gaps that allows you to manually choose whether a gap should be listed in the RSS feed to allow some control over what would be grabbed?

**Comment by tuxpeople** [comment]
Interesting question. From my personal point of view, it depends on the use case. My use case evolved to not logging in to Gaps anymore :-) For me, Gaps "just works" and provides its list to Radarr. If Gaps adds a movie to Radarr which I don't want, I delete it from Radarr and select that:

![image](https://user-images.githubusercontent.com/1506076/199272612-6addfba9-5b04-400b-97e5-c8ca1e03ebb1.png)

That way it will not be added again, neither by Gaps or any other list.
